GoMeet/src/test/java/com/google/sps/VoteMeetingTimeServletTest.java

```diff
@@ -0,0 +1,284 @@
+    writer = new PrintWriter(stringWriter);
+    when(mockedResponse.getWriter()).thenReturn(writer);
+    
+    // Create an existing MeetingTime entity to be voted on and store in local datastore
+    Entity meetingTime = new Entity("MeetingTime");
+    meetingTime.setProperty(MeetingTimeFields.DATETIME, DATETIME_VAL);
+    meetingTime.setProperty(MeetingTimeFields.VOTE_COUNT, VOTE_COUNT_VAL);
+    meetingTime.setProperty(MeetingTimeFields.VOTERS, VOTERS_VAL);
+    datastore.put(meetingTime);
+    MEETING_TIME_ID_VAL = KeyFactory.keyToString(meetingTime.getKey());
+  }
+   
+  @Test
+  public void testSuccessfulVoteTime() throws IOException {
+    when(mockedRequest.getParameter(MeetingTimeFields.MEETING_TIME_ID)).thenReturn(MEETING_TIME_ID_VAL); 
+    when(mockedRequest.getParameter(MeetingTimeFields.VOTERS)).thenReturn(NEW_VOTER);
+    new VoteMeetingTimeServlet().doPost(mockedRequest, mockedResponse);
+    
+    // Check that the entity was updated as expected.
+    Entity meetingTime = getHardcodedEntity();
+
+    // voters should now include NEW_VOTER, and the original voters
+    HashSet<String> voters = new HashSet<String>( (ArrayList<String>) meetingTime.getProperty(MeetingTimeFields.VOTERS));
+    assertTrue(voters.contains(NEW_VOTER));
+    assertTrue(voters.containsAll(VOTERS_VAL));
+
+    // vote count should be incremented by 1
+    Long voteCount = (long) meetingTime.getProperty(MeetingTimeFields.VOTE_COUNT);
+    assertTrue(voteCount == VOTE_COUNT_VAL + 1);
+
+    // HttpServletResponse.SC_OK status code should be returned
+    HashMap<String, Object> status = new HashMap<String, Object>() {{
+      put("status", HttpServletResponse.SC_OK);
+    }};
+    writer.flush(); // writer may not have been flushed yet
+    assertNotNull(stringWriter.toString()); 
+    assertTrue(stringWriter.toString().contains(ServletUtil.convertMapToJson(status)));
+  }
+  
+  @Test 
+  public void testNullMeetingTimeId() throws IOException {
+    when(mockedRequest.getParameter(MeetingTimeFields.MEETING_TIME_ID)).thenReturn(null); // no ID provided
+    when(mockedRequest.getParameter(MeetingTimeFields.VOTERS)).thenReturn(NEW_VOTER);
+    new VoteMeetingTimeServlet().doPost(mockedRequest, mockedResponse);
+
+    // check that error response was sent with appropriate details
+    HashMap errorResponse = new HashMap<String, Object>() {{
+      put("status", HttpServletResponse.SC_BAD_REQUEST);
+      put("message", ErrorMessages.BAD_REQUEST_ERROR);
+    }};
+    String errorJson = ServletUtil.convertMapToJson(errorResponse);
+    // verify status code set
+    verify(mockedResponse, times(1)).setStatus(HttpServletResponse.SC_BAD_REQUEST);
+
+    // Check that nothing was modified 
+    assertTrue(checkNothingModified());
+
+    // Assert that error response is returned
+    writer.flush(); // writer may not have been flushed yet
+    assertNotNull(stringWriter.toString());
+    assertTrue(stringWriter.toString().contains(errorJson));
```

Comment:
Optional - since these lines are repeated in each of the error tests, it could be good to abstract that into a helper function. Something similar to what I've done at the bottom of this [test file](https://github.com/googleinterns/step271-2020/pull/34/files#diff-75ddff44f177749d1205dd33040fe302f9350d3ab3a4b488fc3ae1983a45aae0) (copied my comment from PR#32)

Reply:
Yep, I've refactored it. Thanks for the suggestion!